_episodes/13.md

```diff
@@ -0,0 +1,449 @@
+us the green square above from our cube.
+
+```
+SELECT part, store, COUNT(*)
+FROM cube_table
+WHERE customer = "Bob"
+GROUP BY part, store
+```
+
+Now what if we want to flatten one of the dimensions? While this can be managed
+with a GROUP BY as before, but depending on the system may ignore any cached
+data and scan over all the rows. For this, SQL reserved a special set of
+keywords around cubes. We won't dive into that in depth now but for our current
+goal of flattening a dimension, we can use ROLLUP. Using the keyword ROLLUP
+indicates to the underlying system that you intend to aggregate over the 
+pre-materialized data rather than scan over all rows to compute again. This will
+give you the total count of parts per store using the counts of the data cube.
+
+```
+SELECT part, store, COUNT(*)
+FROM cube_table
+GROUP BY ROLLUP (part, store)
+```
+
+Now, although we used simple counts, you can precompute a lot of other aggregate
+data like sums, min, max, percentile, etc... These can service various queries
+that are commonly queried and doesn't require a new computation every time. That
+is the goal of MOLAP and data cubes.
+
+### Apache Pinot
+
+Now let's move on to Apache Pinot is a Realtime distributed OLAP datastore, 
+designed to answer OLAP queries with low latency. Although there may be a lot of
+words there that overlap with the Trino description, the key differentiators are
+realtime and low latency. Trino performs batch processing and is not a realtime
+system where Pinot is great for ingesting data in batch or stream. The other key
+word, low latency could technically apply to both Pinot and Trino but in the
+context of realtime subsecond latency, Trino is slow compared to Pinot. This
+is due to the specialized indexes that Pinot uses to store the data that we
+will cover shortly. Xiang has a really great summary slide that easily shows
+the strengths of each system and why they work so well together.
+
+<p align="center">
+<img align="center" width="75%" height="100%" src="/assets/episode/13/latency_flexibility.png"/><br/>
+</p>
+
+While Trino is not as fast as Pinot, it is able to handle a broader set of
+use cases like performing broad joins over open data formats in data lakes. 
+This is what motivated work on the Trino Pinot Connector. You can have the speed
+of Pinot, while having the flexibilty of Trino.
+
+Now that you understand the common use case for Pinot, it's important to know 
+the main goals of Pinot. 
+ 1. One primary goal is the keep response times of aggregation queries
+  predictable, regardless of how many requests Pinot handles. As it scales
+  you won't see a degradation of performance. This is achieved by Pinot's
+  custom indices and storage formats. 
+    <p align="center">
+       <img align="center" width="75%" height="100%" src="/assets/episode/13/data_value.jpeg"/><br/>
+    </p>
+
+ 2. Another goal of Pinot is to revive the value of data from a historical
```

Comment:
revive ?

Reply:
One of Xiang's points is that data loses it's value as time goes on, especially ones it falls of the grid of fast searching data stores like elasticsearch/redis/mongo etc.. Using Pinot as a fast way to serve aggregate data revives that value that would otherwise be lost in historic data vs having to scan over all the old data every time to get it's value.

We don't have to use revive but that's the jist of what I'm trying to explain.

Reply:
increase the value of historical data

or something like that maybe